### PR TITLE
[MM-52618] Adding generic error to handle network issues when fetching groups

### PR DESCRIPTION
**Pull request by BenCookie95**
#### Summary
This will handle instances where our ldap groups endpoint times out. Unfortunately with golang there is no way to return an error for an endpoint timing out so we have to make it generic. You can read more about this here:

https://github.com/golang/go/issues/21389

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-52618

#### Screenshots
Screenshot 2023-05-24 at 1 44 02 PM

#### Release Note

```release-note
None
```

### CODE REVIEW

**Review by JulienTant — APPROVED** [review]
Looks good @BenCookie95!

File: webapp/channels/src/components/admin_console/group_settings/groups_list/groups_list.tsx [hunk 1]
```diff
@@ -434,9 +442,16 @@ export default class GroupsList extends React.PureComponent<Props, State> {
...
         };
         this.setState(newState as any);
-        this.props.actions.getLdapGroups(this.state.page, LDAP_GROUPS_PAGE_SIZE, {q: ''}).then(() => {
-            this.setState({loading: false});
-        });
+        this.props.actions.getLdapGroups(this.state.page, LDAP_GROUPS_PAGE_SIZE, {q: ''}).then(this.handleGetGroupsResponse);
+    };
+
+    handleGetGroupsResponse = (response: any) => {
+        if (response.error) {
```
Comment (pvev) [inline webapp/channels/src/components/admin_console/group_settings/groups_list/groups_list.tsx:449]: the conditional part could better be defined in a simpler way as:  `this.setState({fetchError: Boolean(response.error)});`

**Review by pvev — APPROVED** [review]
LGTM! thanks @BenCookie95 👍

**Review by yasserfaraazkhan — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]
thank you for the changes.

**Pull request merged**